complex.js

```diff
@@ -130,6 +128,8 @@
 
   var parse = function(a, b) {
 
+    var P = {'re': 0, 'im': 0};
+
```

Comment:
Should the `P` variable be changed to lowercase now that is is no longer a global?  Perhaps change it to `z` to reflect its purpose?